internal/watcher/p2p_msgs.go

```diff
@@ -52,36 +56,37 @@ func NewP2PMsgsWatcherWatcher(
 		recentBlocksMux:       sync.Mutex{},
 		recentBlocks:          make(map[uint64]*BlockReceivedEvent),
 		mostRecentBlock:       0,
+		txMapper:              txMapper,
 	}
 }
 
-func (dkw *P2PMsgsWatcher) Start(ctx context.Context, runner service.Runner) error {
-	p2pService, err := p2p.New(dkw.config.P2P)
+func (pmw *P2PMsgsWatcher) Start(ctx context.Context, runner service.Runner) error {
+	p2pService, err := p2p.New(pmw.config.P2P)
 	if err != nil {
 		return err
 	}
-	p2pService.AddMessageHandler(dkw)
+	p2pService.AddMessageHandler(pmw)
 
-	runner.Go(func() error { return dkw.insertBlocks(ctx) })
+	runner.Go(func() error { return pmw.insertBlocks(ctx) })
 
 	return runner.StartService(p2pService)
 }
 
-func (dkw *P2PMsgsWatcher) MessagePrototypes() []p2pmsg.Message {
+func (pmw *P2PMsgsWatcher) MessagePrototypes() []p2pmsg.Message {
 	return []p2pmsg.Message{
 		&p2pmsg.DecryptionKeys{},
 		&p2pmsg.DecryptionKeyShares{},
 	}
 }
 
-func (dkw *P2PMsgsWatcher) ValidateMessage(_ context.Context, msgUntyped p2pmsg.Message) (pubsub.ValidationResult, error) {
+func (pmw *P2PMsgsWatcher) ValidateMessage(_ context.Context, msgUntyped p2pmsg.Message) (pubsub.ValidationResult, error) {
```

Comment:
I think we should validate the messages a bit more, otherwise fake messages can easily mess with the metrics. But this can be left for another PR

Reply:
alright we can do that